modules/volume/src/processors/volumeregionmap.cpp

```diff
@@ -0,0 +1,116 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/volume/processors/volumeregionmap.h>
+#include <modules/base/algorithm/volume/volumevoronoi.h>
+
+#include <algorithm>
+#include <functional>
+#include <iostream>
+#include <fstream>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeRegionMap::processorInfo_{
+    "org.inviwo.VolumeRegionMap",                               // Class identifier
+    "Volume Region Map",                                        // Display name
+    "Volume Operation",                                         // Category
+    CodeState::Stable,                                          // Code state
+    Tag::CPU | Tag{"Volume"} | Tag{"Atlas"} | Tag{"DataFrame"}  // Tags
+};
+const ProcessorInfo VolumeRegionMap::getProcessorInfo() const { return processorInfo_; }
+
+VolumeRegionMap::VolumeRegionMap()
+    : Processor()
+    , dataFrame_("mapping_indexes")
+    , inport_("inport")
+    , outport_("outport")
+    , from_{"from", "From", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 1}
+    , to_{"to", "to", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 2} {
+
+    addPort(inport_);
+    addPort(dataFrame_);
+    addPort(outport_);
+    addProperties(from_, to_);
+}
+
+namespace {
+constexpr auto copyColumn = [](const Column& col, auto& dstContainer, auto assign) {
+    col.getBuffer()->getRepresentation<BufferRAM>()->dispatch<void, dispatching::filter::Scalars>(
+        [&](auto buf) {
+            const auto& data = buf->getDataContainer();
+            for (auto&& [src, dst] : util::zip(data, dstContainer)) {
+                std::invoke(assign, src, dst);
+            }
+        });
+};
+}
+
+void VolumeRegionMap::process() {
+    // Get columns
+    auto csv = dataFrame_.getData();
+    auto oldIdx = csv->getColumn(from_.getSelectedValue());
+    auto newIdx = csv->getColumn(to_.getSelectedValue());
+    auto nrows = csv->getNumberOfRows();
+
+    // Store in vectors
+    std::vector<uint32_t> sourceIndices(nrows);
+    std::vector<uint32_t> destinationIndices(nrows);
+
+    copyColumn(*oldIdx, sourceIndices,
+               [](const auto& src, auto& dst) { dst = static_cast<uint32_t>(src); });
```

Comment:
Should uint32_t be assumed?
Possible to use the type of the inport data?

Reply:
The input type is int as of now. I discussed this with Martin and we figured it makes the most sense if you would like to set something to a negative value. You can now specify if voxel values that are not in the source/destination file should be set to a given value in the GUI or keep the current value.